src/System.Runtime.InteropServices.RuntimeInformation/src/RuntimeInformation.Windows.cs

```diff
@@ -1,13 +1,77 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Diagnostics;
+
 namespace System.Runtime.InteropServices
 {
-    public static class RuntimeInformation
+    public static partial class RuntimeInformation
     {
+        private static string s_osDebugName = null;
+        private static Architecture? s_osArch = null;
+
         public static bool IsOSPlatform(OSPlatform osPlatform)
         {
             return OSPlatform.Windows == osPlatform;
         }
+
+        public static string GetOSDebugName()
+        {
+            if (null == s_osDebugName)
+            {
+                string type = Interop.mincore.IsWindowsServer() ? "Server" : "Client";
+                string version = "Unknown";
+
+                if (Interop.mincore.IsWindows10OrGreater())
+                    version = "10";
```

Comment:
I don't understand how the new Windows APIs make the situation any better.  They provide the ability to compare a version you provide against the actual version.  So can't I write this function (using Priya's P/Invokes)?
```C#
private static string GetVersion()
{
    for (uint major = 0; ; major++)
    {
        if (IsVersionOrGreater(major, 0) && !IsVersionOrGreater(major + 1, 0))
        {
            for (uint minor = 0; ; minor++)
            {
                if (IsVersionOrGreater(major, minor) && !IsVersionOrGreater(major, minor + 1))
                {
                    return major + "." + minor;
                }
            }
        }
    }
}
```
If GetVersion lies, does VerifyVersionInfo also lie?  If no, then why not use VerifyVersionInfo in a spirit similar to what I've done here and actually give back meaningful info to the caller.  If yes, then why are we jumping through hoops to avoid using GetVersion when we're not going to be giving back any better data from VerifyVersionInfo?  This all comes back to my original question/comment: why not just give back whatever information the OS has for us, rather than trying to pass it through some kind of filter, especially if we're going to need to rev that filter every time the OS revs?